script.twittertweets/service.py

```diff
@@ -167,30 +162,47 @@ def chkFEED(self, user):
+            except: tmpImage = None
+
             try:
-                tmpAttach   = tweetMsgs[idx]('a', {'class': 'twitter-timeline-link u-hidden'})[0].get_text()
-                tweetAttach = tweetAtts[idx].find('img').attrs['src'] if tmpAttach.startswith('pic.twitter.com') else 'NA.png'
-            except:
-                tweetAttach = 'NA.png'
-                
-            tweetTime  = self.correctTime(tweetTimes[idx]["title"].encode("utf-8"))
-            tweetMsg   = self.cleanString((tweetMsgs[idx]('p' , {'class': 'TweetTextSize TweetTextSize--normal js-tweet-text tweet-text'})[0].get_text()).encode("utf-8"))
-            tweetStats = [stat.get_text().encode("utf-8") for stat in tweetStats]
-            tweetStats = [tweetStats[x:x+3] for x in xrange(0, len(tweetStats), 3)]
+                tmpThumb    = tweetAtts[idx]('div', {'class': 'PlayableMedia-container'})
+                tmpThumb    = tmpThumb[0]('div', {'class': 'PlayableMedia-player'})[0]
+                tmpThumb    = re.findall("background-image:url(.*)'", str(tmpThumb))[0].replace("('",'')
+            except: tmpThumb = None
+            
+            if tmpThumb is not None:
+                tmpVideo    = tmpThumb.replace('pbs.twimg.com/tweet_video_thumb/','video.twimg.com/tweet_video/').replace('.jpg','.mp4')
+            
+            tmpImage    = tmpImage if tmpThumb is None else tmpThumb
+            tweetTime   = self.correctTime(tweetTimes[idx]["title"].encode("utf-8"))
+            tweetMsg    = self.cleanString((tweetMsgs[idx]('p' , {'class': 'TweetTextSize TweetTextSize--normal js-tweet-text tweet-text'})[0].get_text()).encode("utf-8"))
+            tweetStats  = [stat.get_text().encode("utf-8") for stat in tweetStats]
+            tweetStats  = [tweetStats[x:x+3] for x in xrange(0, len(tweetStats), 3)]
             tweetStats = tweetStats[idx]
-
+            if len(tweetMsg) == 0: return
+            
+            # test
+            # tmpImage = 'https://pbs.twimg.com/tweet_video_thumb/DOubaqGWAAEBPWn.jpg'
+            # tmpVideo = 'https://video.twimg.com/tweet_video/DOubaqGWAAEBPWn.mp4'
+            # test
+            
             if getProperty('%s.%s.time' %(ADDON_ID,user)) != tweetTime:
                 setProperty('%s.%s.time'%(ADDON_ID,user),tweetTime)
-                ui = gui.GUI("%s.default.xml" %ADDON_ID,ADDON_PATH,"default",params=({'user':user,'attachment':tweetAttach,'icon':twitterPic,'username':twitterAlt,'title':tweetMsg,'time':tweetTime,'stats':tweetStats,'verified':twitterVer}))
+                ui = gui.GUI("%s.default.xml" %ADDON_ID,ADDON_PATH,"default",params=({'user':user,'video':tmpVideo,'image':tmpImage,'icon':twitterPic,'username':twitterAlt,'title':tweetMsg,'time':tweetTime,'stats':tweetStats,'verified':twitterVer}))
                 ui.doModal()
-        except Exception,e:
-            log('chkFEED, failed! ' + str(e))
-            return
+        except Exception,e: log('chkFEED, failed! ' + str(e))
```

Comment:
`except Exception as e` for python3 compatibility

Reply:
Sorry, Going to have to train my eye for that one... I've spent years using this syntax :)

Reply:
Fixed 👍